service/src/main/kotlin/fi/espoo/evaka/shared/job/JobSchedule.kt

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.shared.job
+
+import com.github.kagkarlsson.scheduler.task.schedule.CronSchedule
+import com.github.kagkarlsson.scheduler.task.schedule.Daily
+import com.github.kagkarlsson.scheduler.task.schedule.Schedule
+import fi.espoo.evaka.application.utils.helsinkiZone
+import java.time.LocalTime
+
+interface JobSchedule {
+    fun getScheduleForJob(job: ScheduledJob): Schedule?
+
+    companion object {
+        fun daily(at: LocalTime): Schedule = Daily(helsinkiZone, at)
+        fun cron(expression: String): Schedule = CronSchedule(expression, helsinkiZone)
+    }
+}
+
+class DefaultJobSchedule : JobSchedule {
+    override fun getScheduleForJob(job: ScheduledJob): Schedule? = when (job) {
+        ScheduledJob.VardaUpdate -> JobSchedule.daily(LocalTime.of(23, 0))
+        ScheduledJob.EndOfDayAttendanceUpkeep -> JobSchedule.daily(LocalTime.of(0, 0))
+        ScheduledJob.KoskiUpdate -> JobSchedule.daily(LocalTime.of(0, 0))
+        ScheduledJob.RemoveOldDraftApplications -> JobSchedule.daily(LocalTime.of(0, 30))
+        ScheduledJob.CancelOutdatedTransferApplications -> JobSchedule.daily(LocalTime.of(0, 35))
+        ScheduledJob.EndOutdatedVoucherValueDecisions -> JobSchedule.daily(LocalTime.of(1, 0))
+        ScheduledJob.RemoveOldAsyncJobs -> JobSchedule.daily(LocalTime.of(3, 0))
+        ScheduledJob.DvvUpdate -> JobSchedule.daily(LocalTime.of(6, 0))
+        ScheduledJob.RemoveOldDaycareDailyNotes -> JobSchedule.daily(LocalTime.of(6, 30))
+        ScheduledJob.SendPendingDecisionReminderEmails -> JobSchedule.daily(LocalTime.of(7, 0))
```

Comment:
Would these schedules be overridable in other implementations (i.e. trevaka)? These times can be highly dependent on the deployment environment (e.g. in Espoo, we have a specific downtime for VTJ access which forces `DvvUpdate` to be at a specific time).

If not as-is (I haven't yet familiarized myself with how the fork-system works for evaka-service), then maybe these should be env vars with defaults?

Reply:
This is just the default configuration, and in practice a `JobSchedule` implementation needs in any case to be provided before the service can start (in our case it's in `EspooConfig.kt`). It's up to the implementation to decide whether it uses this default configuration, uses some mix of default + overrides, or pulls values from env vars.